test/disk_uploader_test.go

```diff
@@ -0,0 +1,127 @@
+	"context"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/utils/ptr"
+
+	v1 "kubevirt.io/api/core/v1"
+	kubevirtcliv1 "kubevirt.io/client-go/kubecli"
+	cdiv1beta1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/constants"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/framework"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/runner"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/testconfigs"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/vm"
+)
+
+var _ = Describe("Run disk-uploader", func() {
+	f := framework.NewFramework()
+
+	BeforeEach(func() {
+		if f.TestOptions.SkipDiskUploaderTests {
+			Skip("skipDiskUploaderTests is set to true, skipping tests")
+		}
+	})
+
+	It("Extracts disk from VM and upload to container registry", func() {
+		imageDestination := os.Getenv("IN_CLUSTER_IMAGE_REGISTRY")
+		Expect(imageDestination).ToNot(BeEmpty())
+
+		registryCredentials, err := createRegistryCredentials(f.CoreV1Client, f.DeployNamespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		cirrosDataVolume := newCirrosDataVolume()
+		cirrosVm, err := createCirrosVM(f.KubevirtClient, f.DeployNamespace, cirrosDataVolume)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = vm.WaitForVM(f.KubevirtClient, f.DeployNamespace, cirrosVm.Name, "", constants.Timeouts.WaitForVMStart.Duration, false)
+		Expect(err).ToNot(HaveOccurred())
+
+		config := &testconfigs.DiskUploaderTestConfig{
+			TaskRunTestConfig: testconfigs.TaskRunTestConfig{},
+			TaskData: testconfigs.DiskUploaderTaskData{
+				ExportSourceKind: "vm",
+				ExportSourceName: cirrosVm.Name,
+				VolumeName:       cirrosVm.Spec.Template.Spec.Volumes[0].DataVolume.Name,
+				ImageDestination: imageDestination,
+				SecretName:       registryCredentials.Name,
+			},
+		}
+		f.TestSetup(config)
+
+		runner.NewTaskRunRunner(f, config.GetTaskRun()).
+			CreateTaskRun().
+			ExpectSuccess().
+			ExpectLogs(config.GetAllExpectedLogs()...).
+			ExpectResults(nil)
+	})
+})
+
+func createRegistryCredentials(client kubeclientcorev1.CoreV1Interface, namespace string) (*corev1.Secret, error) {
+	registryKeyId := os.Getenv("IN_CLUSTER_REGISTRY_ACCESS_KEY_ID")
+	Expect(registryKeyId).ToNot(BeEmpty())
+
+	registryKey := os.Getenv("IN_CLUSTER_REGISTRY_SECRET_KEY")
+	Expect(registryKey).ToNot(BeEmpty())
+
+	v1Secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: constants.E2ETestsRandomName("disk-uploader-credentials"),
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"accessKeyId": []byte(registryKeyId),
+			"secretKey":   []byte(registryKey),
+		},
+	}
+
+	return client.Secrets(namespace).Create(context.Background(), v1Secret, metav1.CreateOptions{})
```

Comment:
@0xFelix @ksimon1 This secret is still required even with OCP's in-cluster registry. If there is no secret, and no secret-name passed to the Tekton's task run, then it would fail to get the `accessKeyId` and `secretKey` from `secretKeyRef` [1].

[1] https://github.com/kubevirt/kubevirt-tekton-tasks/blob/main/release/tasks/disk-uploader/disk-uploader.yaml#L48-L57

Reply:
But in this case it should not be a problem to store credentials inside the tests, since the internal registry of the test cluster is accessible only from within the cluster and not from outside.